README.md

```diff
@@ -1,9 +1,170 @@
 
 ## Install
 
 ```sh
 brew install takescoop/formulas/kubectl-exec-forward
 ```
+
+## Usage
+
+```sh
+kubectl exec-forward type/name port [flags] [-- command]
+```
+
+### Flags
+
+In addition to the standard kubectl flags, the following flags are available for the plugin
+
+| Flag | Short | Description | Default |
+|---|---|---|---|
+| `--arg` | `-a` | `key=value` arguments passed to commands | `[]` |
+| `--verbose` |`-v`| Whether to log verbosely |`false` |
+| `--pod-timeout` | `-t` | Time to wait for an attachable pod to become available | `500` (ms) |
+| `--persist` | `-p` | Whether to persist the forwarding connection after the main command has finished | `false` |
+
+## Administration
+
+Administrators can store complex behavior in Kubernetes pod annotations, allowing users to run a single `kubectl` to interact with remote resources.
+
+### Lifecycle
+
+| Name | Description | 
+|---|---|
+| `pre-connect` | Run before establishing a port-forwarding connection |
+| `post-connect` | Run after establishing a port-forwarding connection |
+| `command` | The main command, run after `post-connect`. When `command` finishes, the port-forwarding connection is closed | 
+
+### Annotations
+
+| Name | Description | 
+|---|---|
+| `exec-forward.pod.kubernetes.io/args` | Arguments passed to commands for rendering, can be overridden from the CLI via `--arg\|-a` |
+| `exec-forward.pod.kubernetes.io/pre-connect` | A JSON formatted list of commands executed before establishing a port-forwarding connection |
+| `exec-forward.pod.kubernetes.io/post-connect` | A JSON formatted list of commands executed after establishing a port-forwarding connection |
+| `exec-forward.pod.kubernetes.io/command` | A single JSON formatted command ran after `post-connect` |
+
+#### Command
+
+##### Object
+
+The command object represents an exec command and associated configuration to be executed by the plugin during the port-forwarding lifecycle.
+
+| Attribute | Description | Optional |
+| --- | --- | --- |
+| `id` | A unique identifier that can be used in subsequent commands to reference a previous command's output | `true` |
+| `command` | The command to run as an array of strings | `false` |
+| `interactive` | Whether the command should be run in interactive mode. Default is `false`. Note: the main `command` is always run in interactive mode | `true` |
+| `name` | The display name for the command, shown during execution | `false` |
+
+##### Rendering
+
+Exec commands are [Go templated](https://pkg.go.dev/text/template) using inputs from the CLI, the `args` annotation, configuration from the port-forwarding connection and previous command outputs.
+
+| Namespace | Description | Example |
+|---|---|---|
+| `.Args` | Arguments read from the `args` annotation and overridden using the `--arg\|-a` CLI flags | `{{.Args.username}}` |
+| `.Outputs` | Stdout and Stderr from previously ran commands, stored by command `id` | `{{.Outputs.foo.Stdout}}` |
```

Comment:
Outputs could probably just be stdout straight up, i.e. `.Outputs.foo`. Stderr can be stored to print errors but doesn't need to be available for command templating.

Reply:
yeah i was thinking about this. I'll open a separate PR.